src/utils/canonicalJson.ts

```diff
@@ -0,0 +1,46 @@
+export type JsonPrimitive = null | boolean | number | string;
+export type JsonValue =
+  | JsonPrimitive
+  | JsonValue[]
+  | { [key: string]: JsonValue };
+
+export type RequestPayload<Params = Record<string, unknown>> = {
+  method: string;
+  params: Params;
+};
+
+function isPlainObject(value: unknown): value is Record<string, unknown> {
+  return (
+    typeof value === 'object' &&
+    value !== null &&
+    Object.getPrototypeOf(value) === Object.prototype
+  );
+}
+
+function canonicalize(value: unknown): unknown {
```

Comment:
curious if we could leave comments on the usage of `canonical` - it could have various meanings